### PR TITLE
Better macros

### PR DESCRIPTION
**Pull request by FredTheDino**
- Add MatchableValue
- Rewrite lib sylt
- Make the macro even more prettier
- Minor fixes

### CODE REVIEW

**Comment by FredTheDino** [comment]
It's the doc-test that fails... No idea why. One solution is changing the CI to `cargo test prog`.

It's frustrating tough -- the compiler complains about it doesn't understand what is what. But it's apparently not a problem when it's compiling. I almost suspect we've hit an edge case in the compiler.

**Comment by sornas** [comment]
The compiler suggests two alternatives:

```
    = note: `sylt` could refer to a crate passed with `--extern`
    = help: use `::sylt` to refer to this crate unambiguously
note: `sylt` could also refer to the module imported here
   --> /home/gustav/dev/sylt/src/lib_sylt.rs:2:5
    |
2   | use crate as sylt;
    |     ^^^^^^^^^^^^^
    = help: use `self::sylt` to refer to this module unambiguously
```

and applying the second

```
diff --git a/sylt_macro/src/lib.rs b/sylt_macro/src/lib.rs
index 2649b25..b577016 100644
--- a/sylt_macro/src/lib.rs
+++ b/sylt_macro/src/lib.rs
@@ -77,7 +77,7 @@ pub fn extern_function(tokens: proc_macro::TokenStream) -> proc_macro::TokenStre
         let pat = block.pattern.clone();
         let ty = block.return_ty.clone();
         quote! {
-            #pat => { Ok(sylt::Value::from(#ty)) }
+            #pat => { Ok(self::sylt::Value::from(#ty)) }
         }
     }).collect();
 
@@ -95,27 +95,27 @@ pub fn extern_function(tokens: proc_macro::TokenStream) -> proc_macro::TokenStre
         pub fn #function (
             __values: &[sylt::Value],
             __typecheck: bool
-        ) -> ::std::result::Result<sylt::Value, sylt::error::RuntimeError>
+        ) -> ::std::result::Result<self::sylt::Value, self::sylt::error::RuntimeError>
         {
-            use sylt::Value::*;
-            use sylt::MatchableValue::*;
+            use self::sylt::Value::*;
+            use self::sylt::MatchableValue::*;
             if __typecheck {
                 let matching: Vec<_> = __values.iter().map(make_matchable).collect();
                 #[allow(unused_variables)]
                 match matching.as_slice() {
                     #(#typecheck_blocks),*
-                    _ => Err(sylt::error::RuntimeError::ExternTypeMismatch(
+                    _ => Err(self::sylt::error::RuntimeError::ExternTypeMismatch(
                         stringify!(#function).to_string(),
-                        __values.iter().map(|v| sylt::Type::from(v)).collect()
+                        __values.iter().map(|v| self::sylt::Type::from(v)).collect()
                     ))
                 }
             } else {
                 let matching: Vec<_> = __values.iter().map(make_matchable).collect();
                 match matching.as_slice() {
                     #(#eval_blocks),*
-                    _ => Err(sylt::error::RuntimeError::ExternTypeMismatch(
+                    _ => Err(self::sylt::error::RuntimeError::ExternTypeMismatch(
                         stringify!(#function).to_string(),
-                        __values.iter().map(|v| sylt::Type::from(v)).collect()
+                        __values.iter().map(|v| self::sylt::Type::from(v)).collect()
                     ))
                 }
             }
```
makes the tests pass.
